src/structures/GuildChannel.ts

```diff
@@ -0,0 +1,136 @@
+import type { channel, Snowflake } from "../discord.ts";
+import type { Client } from "../Client.ts";
+import { permissionMap } from "./Role.ts";
+import { SnowflakeBase } from "./Base.ts";
+
+export interface PermissionOverwrite {
+  id: Snowflake;
+  type: "role" | "member";
+  permissions: Record<keyof typeof permissionMap, boolean | undefined>;
+}
+
+function parsePermissionOverwrite(allow: string, deny: string) {
+  const bAllow = BigInt(allow);
+  const bDeny = BigInt(deny);
+
+  const permissions = {} as Record<
+    keyof typeof permissionMap,
+    boolean | undefined
+  >;
+
+  for (const [key, val] of Object.entries(permissionMap)) {
+    const bVal = BigInt(val);
+    if ((bAllow & bVal) == bVal) {
+      permissions[key as keyof typeof permissionMap] = true;
+    } else if ((bDeny & bVal) == bVal) {
+      permissions[key as keyof typeof permissionMap] = false;
+    } else {
+      permissions[key as keyof typeof permissionMap] = undefined;
+    }
```

Comment:
Is the equality expressions supposed to be non-strict equality?

Reply:
good catch